### PR TITLE
[11.0][ADD] medical_turn_management

### PR DESCRIPTION
**Pull request by etobella**

### CODE REVIEW

File: medical_turn_management/models/medical_turn.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        return super()._compute_display_name()
+
+    @api.multi
+    def name_get(self):
+        result = []
+        for rec in self:
+            name = '%s [%s] (%s)' % (
+                rec.practitioner_id.display_name or _('Pending to assign'),
+                rec.specialty_id.name,
+                ','.join(c.ref for c in rec.center_ids)
```
Comment (jarroyomorales) [inline medical_turn_management/models/medical_turn.py]: Fails if a center doesnt have `ref`. I know it shouldnt happen but still...

File: medical_turn_management/models/medical_turn_specialty.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        inverse_name='turn_specialty_id',
+    )
+
+    def _execute_rules(self, start_date, end_date):
+        results = self.env['medical.turn']
+        rules = self.mapped('rule_ids')
+        for day in range(0, (end_date-start_date).days):
+            date = start_date + timedelta(days=day)
+            for rule in rules:
+                if date.weekday() != int(rule.dayofweek):
```
Comment (jarroyomorales) [inline medical_turn_management/models/medical_turn_specialty.py:29]: Can be done with a lambda function

**Review by jarroyomorales — APPROVED** [review]
LGTM! Just one question. Why are you not generating the turns with a cron?

File: medical_turn_management/views/medical_turn_specialty.xml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                <sheet>
+                    <group>
+                        <field name="name"/>
+                    </group>
+                    <notebook>
+                        <page name="rule" string="Rules">
+                            <field name="rule_ids">
+                                <tree editable="bottom">
+                                    <field name="dayofweek"/>
+                                    <field name="start_hour"/>
```
Comment (jarroyomorales) [inline medical_turn_management/views/medical_turn_specialty.xml]: widget float_time

File: medical_turn_management/models/medical_turn_specialty.py (hunk 2 above)
Comment (etobella) [reply medical_turn_management/models/medical_turn_specialty.py:29]: Faster this, way, If we use lambda, we will do two fors. Now it's just one.

**Pull request merged**